app/controllers/bus_stops_controller.rb

```diff
@@ -0,0 +1,47 @@
+class BusStopsController < ApplicationController
+  def index
+    @bus_route
+    @bus_stops = bus_route.bus_stops
+  end
+
+  def show
+    @bus_stop
+    raw_predictions = fetch_predictions
+    @predictions = format(raw_predictions)
+  end
+
+  private
+  def bus_route
+    @bus_route ||= BusRoute.find(params.require(:bus_route_id))
+  end
+
+  def bus_stop
+    @bus_stop ||= BusStop.find(params.require(:id))
+  end
+
+  def fetch_predictions
+    BusPredictionsFetcher.execute(bus_route, bus_stop)
+  end
+
+  # THIS IS AN AWFUL WIP
```

Comment:
I didn't say it 😁

Reply:
Working on a better solution to normalize the json response.